### PR TITLE
Allow model columns to bear the same name as reserved wtforms.BaseForm attributes

### LINKED ISSUE #656: Can't save with a ModelView to a Model that has a column named data
### Checklist

- [X] The bug is reproducible against the latest release or `master`.
- [X] There are no similar issues or pull requests to fix it yet.

### Describe the bug

If you create a model with a column named Data, and create an empty ModelView to edit it, then when trying to save a new record you will get an exception, or one very similar to it, see below

### Steps to reproduce the bug

Create a model with an id column and a data column:

```
class MyModel(Base):
    __tablename__ = "mymodels"
    id: Mapped[str] = mapped_column(String, primary_key=True, default=keygen)
    data: Mapped[str] = mapped_column(String, server_default="", nullable=False, default="")

```
Create an sql ModelView to edit it:

```
class MyModelAdmin(ModelView, model=MyModel):
    pass

```
And try to create a new object using the admin page.

Note, if you rename the column from data to something else, the bug bad behavior doesn't happen.

### Expected behavior

No exception, a new record is saved

### Actual behavior

You get an exception that shows an error after clicking save.

### Debugging material

```
'StringField' object has no attribute 'items'
Traceback (most recent call last):
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/sqladmin/application.py", line 570, in edit
    obj = await model_view.update_model(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/sqladmin/models.py", line 978, in update_model
    return await Query(self).update(pk, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/sqladmin/_queries.py", line 221, in update
    return await anyio.to_thread.run_sync(self._update_sync, pk, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2106, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 833, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/sqladmin/_queries.py", line 139, in _update_sync
    obj = self._set_attributes_sync(session, obj, data)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imrigoldberg/Library/Caches/pypoetry/virtualenvs/orch-backend-Yxz7Ridj-py3.11/lib/python3.11/site-packages/sqladmin/_queries.py", line 73, in _set_attributes_sync
    for key, value in data.items():
                      ^^^^^^^^^^
AttributeError: 'StringField' object has no attribute 'items'
```

### Environment

Python 3.11 

Output of pip freeze:

alembic==1.12.0
annotated-types==0.6.0
anyio==4.0.0
appnope==0.1.3
astroid==2.15.8
asttokens==2.4.0
asyncpg==0.28.0
backcall==0.2.0
bcrypt==4.0.1
black==23.10.0
certifi==2023.7.22
cffi==1.16.0
charset-normalizer==3.3.0
click==8.1.7
cryptography==41.0.4
decorator==5.1.1
dill==0.3.7
executing==2.0.0
fastapi==0.100.1
greenlet==3.0.0
h11==0.14.0
httpcore==0.18.0
httpx==0.25.0
idna==3.4
iniconfig==2.0.0
ipython==8.16.1
isort==5.12.0
itsdangerous==2.1.2
jedi==0.19.1
Jinja2==3.1.2
lazy-object-proxy==1.9.0
Mako==1.2.4
MarkupSafe==2.1.3
matplotlib-inline==0.1.6
mccabe==0.7.0
mypy==1.6.1
mypy-extensions==1.0.0
-e git+https://github.com/portpower/orchestration.git@44fb0754b05410942755f965229a65bf81bc7e04#egg=orch_backend&subdirectory=backend
packaging==23.2
parso==0.8.3
passlib==1.7.4
pathspec==0.11.2
pexpect==4.8.0
pickleshare==0.7.5
platformdirs==3.11.0
pluggy==1.3.0
prompt-toolkit==3.0.39
psycopg2-binary==2.9.9
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
pydantic==2.4.2
pydantic_core==2.10.1
Pygments==2.16.1
PyJWT==2.8.0
pylint==2.17.7
pytest==7.4.2
python-dateutil==2.8.2
python-multipart==0.0.6
pytz==2023.3.post1
requests==2.31.0
sentry-sdk==1.32.0
six==1.16.0
sniffio==1.3.0
sqladmin==0.15.2
SQLAlchemy==2.0.22
SQLAlchemy-Utils==0.41.1
stack-data==0.6.3
starlette==0.27.0
tomlkit==0.12.1
traitlets==5.11.2
types-passlib==1.7.7.13
types-python-dateutil==2.8.19.14
types-pytz==2023.3.1.1
types-requests==2.31.0.10
typing_extensions==4.8.0
urllib3==2.0.7
uvicorn==0.23.2
wcwidth==0.2.8
wrapt==1.15.0
WTForms==3.0.1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brouberol**
For example, a Model with a mapped column named `data` would collide with the [`wtforms.Form.data`](https://github.com/aminalaee/sqladmin/blob/main/sqladmin/forms.py#L619-L634) property, which would generate errors when processing or rendering the form, as reported in #656.

What we do here is silently rename the column when entering and leaving WTForms territory. Taking the example of a column named `data`, the associated `Field` object will:
- be listed under the form key `data_`
- be passed a `name='data'` kwarg so that it is still displayed with the `data` label in the admin web UI

Instead of directly passing `form.data` to `model_view.insert_model` or `model_view.update_model`, we first restore the original name (e.g. `data_` to `data`) so that the underlying SQLAlchemy can them work on the original column.

> **Note**: this is a bit crude and I think the naming could be improved. Feel free to refactor at will.

Fixes #656

### CODE REVIEW

File: sqladmin/forms.py [hunk 1]
```diff
@@ -92,6 +93,17 @@ def __call__(
 
 T_CC = TypeVar("T_CC", bound=ConverterCallable)
 
+# If the model has a mapped column named e.g. 'data', this will utltimately
+# shadow the wtforms.Form.data attribute, thus causing issues
+# (see https://github.com/aminalaee/sqladmin/issues/656).
+# To circumvent the issue, we maintain a list of reserved attribute names that
+# sqlmodel will silently rename between when converting a sqlalchemy mapped column
```
Comment (aminalaee) [inline sqladmin/forms.py:100]: I think you meant `sqladmin`

File: sqladmin/application.py [hunk 2]
```diff
@@ -509,8 +513,14 @@ async def create(self, request: Request) -> Response:
                 request, model_view.create_template, context, status_code=400
             )
 
+        # if the model defines mapped columns that bear the name of WFTforms reserved
```
Comment (aminalaee) [inline sqladmin/application.py:516]: Can we summarize all new comments to one-line only?

**Review by aminalaee — COMMENTED** [review]
Thanks for the PR, I think it looks good, just some cleanup is needed

**Comment by brouberol** [comment]
I've added some tests as well to satisfy the codecov job. Feel free to refactor things around if you want to.

**Review by aminalaee — APPROVED** [review]
Thanks for the contribution 👍

**Pull request merged**